includes/Checker/Checks/Plugin_Repo/File_Type_Check.php

```diff
@@ -244,6 +247,63 @@ protected function look_for_application_files( Check_Result $result, array $file
 		}
 	}
 
+	/**
+	 * Looks for application files and amends the given result with an error if found.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 */
+	protected function look_for_badly_named_files( Check_Result $result, array $files ) {
+		$conflict_chars = array( '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '+', '=', '[', ']', '{', '}', ';', ':', '"', '\'', '<', '>', ',', '?', '/', '\\', '|', '`', '~' );
+
+		foreach ( $files as $file ) {
+			$badly_name = false;
+			if ( preg_match('/\s/', $file ) ) {
+				$badly_name = true;
+			}
+
+			foreach ( $conflict_chars as $char ) {
+				if ( strpos( basename( $file ) , $char ) !== false ) {
+					$badly_name = true;
+					break;
+				}
+			}
```

Comment:
Instead of a loop you could just use a regex as well.

Something like this (untested):

```suggestion
			if ( preg_match( '/' . preg_quote( implode( $conflict_chars, '|' ), '/' ) . '/' ) ) {
				$badly_name = true;
			}
```

Reply:
Good point! I'll change it then.